src/main/java/io/cryostat/net/web/http/api/v2/graph/AllArchivedRecordingsFetcher.java

```diff
@@ -81,6 +82,16 @@ String name() {
         return "archivedRecordings";
     }
 
+    @Override
+    SecurityContext securityContext(DataFetchingEnvironment environment) {
+        // FIXME what to do here? all users should be allowed to make this request, but should only
```

Comment:
Since this is the All Archives Recordings fetcher, assuming the user has permissions across all visible namespaces, wouldn't it make sense to just try and query all namespaces and match the token with each?

Reply:
The Cryostat instance could be deployed in a multi-tenant scenario where the user who deployed Cryostat is not the only user who will interact with Cryostat. It could be a Platform-as-a-Service situation where admin A owns the cluster and allows users U and V to rent compute resources within namespaces U' and V' to run their own projects, so U has permissions only within U' and V only has permissions within V'. If U makes a request for "all" the archived recordings then the only ones where they have permissions to do that is U', but the cluster may also have namespaces B, C, D, ... Z, and more, so we don't want to enumerate every single namespace and perform an authorization check every time U just tries to list their archives. If we did that then we would have a list of allowed namespaces and then we would need to iterate over all of the archives and filter for only the ones where the security context matches an entry in the allowed namespaces list.

The `/api/v1/recordings` handler has the same problem too.

Doing the sort-of-inverse operation seems slightly better - check all the archived recordings Cryostat has, collect a list (or set) of all the security contexts, and then simply test the user's token against all of those contexts and filter out recordings where this test fails. This way the number of namespaces on the cluster doesn't really matter. It should be a bit more performant this way since we know that the maximum size of the set of recording security contexts is equal to the number of namespaces in the cluster, but the number of namespaces in the cluster can far exceed the number of security contexts present among archived recordings.